src/research_index_backend/doi.py

```diff
@@ -0,0 +1,195 @@
+"""DOI (Digital Object Identifier) validation and management module.
+
+This module handles:
+1. DOI pattern validation
+2. Database existence checks
+3. Metadata validation
+4. Batch processing with limits: TODO
+"""
+
+import time
+from logging import getLogger
+from re import IGNORECASE, compile
+from typing import Dict, List
+
+from neo4j import Driver
+from pydantic import BaseModel
+
+from .session import connect_to_db
+
+logger = getLogger(__name__)
+
+# Use regex pattern from
+# https://www.crossref.org/blog/dois-and-matching-regular-expressions/
+DOI_PATTERN = r"10\.\d{4,9}/(?=.*\d)[-._;()/:A-Z0-9]+$"
+
+class DOI(BaseModel):
+    doi: str
+    valid_pattern: bool = False
+    already_exists: bool = False
+    openalex_metadata: bool = False
+    openaire_metadata: bool = False
+    ingestion_success: bool = False
+
+
+class DOITracker(BaseModel):
+    doi_tracker: Dict[str, DOI]
+
+
+class DOIManager:
+    def __init__(
+        self, list_of_dois: List[str], limit: int, update_metadata=True
+    ) -> None:
+
+        if not list_of_dois:
+            raise ValueError("DOI list cannot be empty")
+        if (limit <= 0) or (limit > len(list_of_dois)):
+            raise ValueError(
+                "Limit must be positive and less than the number of DOIs"
```

Comment:
Error message should read "Limit must be positive and less than _or equal to_ the number of DOIs"